pkg/ui/workspaces/crdb-api-client/BUILD.bazel

```diff
@@ -0,0 +1,84 @@
+load("@aspect_rules_js//js:defs.bzl", "js_binary", "js_run_binary", "js_test")
+load("@aspect_rules_js//npm:defs.bzl", "npm_package")
+load("@aspect_rules_ts//ts:defs.bzl", "ts_project")
+load("@bazel_skylib//rules:write_file.bzl", "write_file")
+load("@npm//:defs.bzl", "npm_link_all_packages")
+load("@npm//:defs.bzl", "npm_link_all_packages")
+load("//docs/generated/http:defs.bzl", "PROTOBUF_TARGETS")
+load("//pkg/ui/workspaces/db-console/src/js:defs.bzl", "proto_sources")
+
+npm_link_all_packages(name = "node_modules")
+
+genrule(
+    name = "protos",
+    srcs = [
+        ":node_modules/@bufbuild/protoc-gen-es/bin/protoc-gen-es",
+        ":proto_sources",
+    ] + PROTOBUF_TARGETS,
+    outs = ["dist"],
+    cmd = """
+    mkdir -p $(OUTS);
+    DESCRIPTOR_SET_IN=$$(for word in $(SRCS); do echo $$word; done | grep '\\.bin$$' | tr -s '\\n' ':');
+    PROTO_PATHS=$$(for f in $(locations :proto_sources);
+        do if [[ "$$f" =~ ^.*/pkg/.*/_virtual_imports/.*_proto/.*.proto$$ ]];
+            then echo $$f | sed 's|.*pkg/.*/_virtual_imports/.*_proto/||' | sed 's|.bin$$||';
+        fi; done | sort -u);
+    export PATH=$$PATH:$$(dirname $(NODE_PATH));
+    $(location @com_google_protobuf//:protoc) \
+    --plugin=$(location :node_modules/@bufbuild/protoc-gen-es/bin/protoc-gen-es) \
```

Comment:
Nit: extra indentation (e.g. 8 spaces instead of 4) on these lines 28-32 would be good to reflect that this is one big command.